### ISSUE TITLE
Git auth failing on "non fork" PR's

### ISSUE THREAD

**Issue by IAmNatch** [wearerequired/lint-action#271]
Hi there,

First off thanks for all your work on this wonderful package, I've really been enjoying working with the linting annotations. 

Unfortunately, I've experienced that any PR that includes checks is failing due to authentication issues. If I merge the PR into my master branch, authentication and all tests pass as expected. I know that there is a separate issue for forked PR's, but these are all PR's within the same root repository, and no forking warning is displayed by the action. 

The error I'm encountering is as follows:
```shell
Adding auth information to Git remote URL
Fetching remote branch "my-awesome-branch"
remote: Invalid username or password.
fatal: Authentication failed for 'https://github.com/my-org/my-repo.git/'
Error: Command failed: git fetch --no-tags --depth=1 origin my-awesome-branch
remote: Invalid username or password.
fatal: Authentication failed for 'https://github.com/my-org/my-repo.git/'
```

A minimum reproduction of my workflow.yml:
```yml
build:
    name: Build
    runs-on: ubuntu-20.04
    steps:
      - name: clone repo
        uses: actions/checkout@v2
      - name: setup node
        uses: actions/setup-node@v2
        with:
          node-version: "14"
          cache: "yarn"
      - name: install dependancies
        run: yarn install
      - name: cache build
        uses: actions/cache@v2
        id: cache-build
        with:
          path: |
            ./*
            /home/runner/.cache/Cypress
          key: ${{ github.sha }}

lint:
    name: Run Linters
    runs-on: ubuntu-20.04
    needs: build
    steps:
      - name: restore build
        uses: actions/cache@v2
        id: restore-build
        with:
          path: |
            ./*
            /home/runner/.cache/Cypress
          key: ${{ github.sha }}
      - name: Run linters
        uses: wearerequired/lint-action@v1
        with:
          # general config
          github_token: ${{ secrets.github_token }}
          continue_on_error: false
          auto_fix: false
          check_name: ${linter} Runner
          # eslint
          eslint: true
          eslint_command_prefix: yarn
          eslint_args: --max-warnings=0
          eslint_extensions: js,jsx,ts,tsx
```

Is there any way to bypass git pull and simply use the directory that is already pulled in by `actions/checkout@v2`?
